code/game/objects/items/robot/cyborg_gripper.dm

```diff
@@ -0,0 +1,223 @@
+			)
+		return FALSE
+	//Already have an item.
+	if(gripped_item)
+
+		//Pass the attack on to the target. This might delete/relocate gripped_item.
+		if(!target.attackby(gripped_item, user, params))
+			// If the attackby didn't resolve or delete the target or gripped_item, afterattack
+			// (Certain things, such as mountable frames, rely on afterattack)
+			gripped_item?.afterattack(target, user, 1, params)
+
+		//If gripped_item either didn't get deleted, or it failed to be transfered to its target
+		if(!gripped_item && length(contents))
+			gripped_item = contents[1]
+			return FALSE
+		else if(gripped_item && !contents.len)
+			gripped_item = null
+
+	//Check that we're not pocketing a mob.
+	else if(isitem(target))
+		var/obj/item/I = target
+		// Make sure the item is something the gripper can hold
+		if(is_type_in_typecache(I, can_hold))
+			to_chat(user, "<span class='notice'>You collect [I].</span>")
+			I.forceMove(src)
+			gripped_item = I
+			return TRUE
+		
+		to_chat(user, "<span class='warning'>Your gripper cannot hold [target].</span>")
+		return FALSE
+
+	// Everything past this point requires being able to engineer.
+	if(!engineering_machine_interaction)
+		return
+
+	if(istype(target, /obj/machinery/power/apc))
+		var/obj/machinery/power/apc/A = target
+		if(A.opened && A.cell)
+			gripped_item = A.cell
+
+			A.cell.add_fingerprint(user)
+			A.cell.update_icon()
+			A.cell.forceMove(src)
+			A.cell = null
+
+			A.charging = APC_NOT_CHARGING
+			A.update_icon()
+
+			user.visible_message("<span class='warning'>[user] removes the power cell from [A]!</span>", "<span class='warning'>You remove the power cell.</span>")
+
+	else if(istype(target, /obj/machinery/cell_charger))
+		var/obj/machinery/cell_charger/cell_charger = target
+		if(cell_charger.charging)
+			gripped_item = cell_charger.charging
+			cell_charger.charging.add_fingerprint(user)
+			cell_charger.charging.forceMove(src)
+			cell_charger.removecell()
+
+	else if(istype(target, /obj/machinery/light))
+		var/obj/machinery/light/light = target
+		var/obj/item/light/L = light.drop_light_tube()
+		L.forceMove(src)
+		gripped_item = L
+		user.visible_message("<span class='notice'>[user] removes the light from the fixture.</span>", "<span class='notice'>You dislodge the light from the fixture.</span>")
+
+	return TRUE
+
+/******************************
+/     UNIVERSAL GRIPPER
+******************************/
+/// Universal gripper. Not supplied to any cyborg by default. Could be varedited onto a borg for event stuff. Functions almost like a real hand!
+/obj/item/gripper/universal		
+	name = "cyborg gripper"
+	desc = "A grasping tool for cyborgs. This one is not restricted by any restraining software, allowing it to handle any object the user wishes."
+	// It's UNIVERSAL so it has all functions enabled.
+	engineering_machine_interaction = TRUE
+	can_help_up = TRUE
+	can_hold = list(
```

Comment:
This is going to be a *very large* typecache. Please make can_hold a method instead, so universal grippers can just return TRUE.

Reply:
Done.